### ISSUE TITLE
Having the ability to set the internal textures filtering

### ISSUE THREAD

**Issue by SkyVault** [treeform/boxy#53]
My usecase for this is for games using pixel art, I have been scaling up my images so that the bi-linear filtering isn't as apparent, but I would like to be able to set the texture filtering to minNearest / magNearest.

**Comment by treeform** [comment]
I also want this, but actually most pixel games don't just set minNearest / magNearest as it looks terrible.

See here and how wavy it looks: https://www.shadertoy.com/view/ltBGWc

One needs to do manual filtering, and there are some tradeoffs.

https://csantosbh.wordpress.com/2014/01/25/manual-texture-filtering-for-pixelated-games-in-webgl/

I have tried to implement something like this but have not solved it yet.

You want to use mipmapping when pixels are small enough, when you zoom in and rotate you want the pixel to look more like little rectangles with AA ages:

![image](https://user-images.githubusercontent.com/125948/219248644-4dac5093-c491-4634-998e-95ca929e3585.png)

That will make pixel art look great.

Scaling up pixel art might be more performant then custom filtering... I would have to test that out.

**Comment by SkyVault** [comment]
Yeah you are correct, this is a very hard problem I've been wanting to find a solution for, I also have a 3d project that uses a pixel art style and the attempts I've made to make that look good have all been pretty bad, but yeah I'm going to see what I can come up with for this project, I also got a lot of interesting reading to do, thanks for the response!

**Comment by treeform** [comment]
What you want here is a special shader that can anti-alias pixel art. Just setting the texture filtering mode will just make everything look even uglier. This is not what you want, but I also think this is outside of what Boxy can do for you.

See https://www.shadertoy.com/view/MlB3D3 or https://www.shadertoy.com/view/mtyXzV 

Sorry, but I don't think I'll be implementing this inside Boxy because I think pixel-perfect pixel art rendering is quite hard and outside the scope of what this is trying to do.
